lib/charliecloud.py

```diff
@@ -1169,6 +1169,13 @@ def root_env():
    def manifest_for_download(self, image_ref):
       return self.download_cache // ("%s.manifest.json" % image_ref.for_path)
 
+   def reset(self):
+      if (os.path.isdir(self.unpack_base) and 
+          os.path.isdir(self.download_cache)):
```

Comment:
Move this predicate into its own function that e.g. returns true or false depending on whether the builder storage directory seems valid.